gateway-topology-hadoop-xml/src/main/java/org/apache/knox/gateway/topology/hadoop/xml/HadoopXmlResourceParser.java

```diff
@@ -266,6 +267,8 @@ private SimpleDescriptor parseXmlDescriptor(String name, String xmlValue) {
         case CONFIG_NAME_PROVIDER_CONFIG_REFERENCE:
           descriptor.setProviderConfig(parameterPairParts[1].trim());
           break;
+        case CONFIG_NAME_PROVISION_ENCRYPT_QUERY_STRING_CREDENTIAL:
+          descriptor.setProvisionEncryptQueryStringCredential(Boolean.valueOf(parameterPairParts[1].trim()));
```

Comment:
A `break` statement is missing here.

Reply:
Indeed; nice catch @zeroflag !
I submitted a new PS.